src/main/java/org/springframework/data/couchbase/repository/ReactiveCouchbaseRepository.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.couchbase.repository;
+
+import java.io.Serializable;
+
+import org.springframework.data.couchbase.core.ReactiveCouchbaseOperations;
+import org.springframework.data.repository.reactive.ReactiveCrudRepository;
+
+/**
+ * @author Subhashni Balakrishnan
+ * @since 3.0
+ */
+public interface ReactiveCouchbaseRepository<T, ID extends Serializable> extends ReactiveCrudRepository<T, ID> {
```

Comment:
If you omit `getCouchbaseOperations()`, then one repository implementation is sufficient. Repository type adoption is handled in Spring Data Commons that adopts the implementation method result into the declared type.